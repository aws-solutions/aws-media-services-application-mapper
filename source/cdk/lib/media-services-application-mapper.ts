/*! Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
       SPDX-License-Identifier: Apache-2.0 */

import {
    Aws,
    Fn,
    Stack,
    StackProps,
    CfnParameter,
} from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { MsamDynamoDB } from './msam-dynamodb';
import { MsamIamRoles } from './msam-iam-roles';
import { MsamEvents } from './msam-events';
import { MsamBrowserApp } from './msam-browser-app'
import { MsamCore } from './msam-core';
import * as utils from './utils';

export class MediaServicesApplicationMapper extends Stack {
    constructor(scope: Construct, id: string, props?: StackProps) {
        super(
            scope,
            id,
            {
                ...props,
                description: '(SO0048) Media Services Application Mapper Root Stack %%VERSION%%',
            }
        );

        /**
         * Parameters
         */
        const cacheItemTTL = new CfnParameter(this, 'CacheItemTTL', {
            default: '86400',
            description: 'This is the maximum time in seconds a cached item will remain if never updated (3600 = 1 hour, 86400 = 1 day, 604800 = 1 week).',
            type: 'String',
            allowedPattern: '\\S+',
            minLength: 1,
            constraintDescription: 'Please enter a value for this field.',
        });

        /**
         * Nested Stacks
         */

        // IAM Module Stack
        const iamModuleStack = new MsamIamRoles(this, 'IAMModuleStack');
        utils.addTemplateUrl(
            iamModuleStack,
            Fn.join('', [
                'https://%%BUCKET_NAME%%-',
                Aws.REGION,
                '.s3.',
                Aws.URL_SUFFIX,
                '/%%SOLUTION_NAME%%/%%VERSION%%/msam-iam-roles-release.template',
            ]),
        );

        // DynamoDB Module Stack
        const dynamoDBModuleStack = new MsamDynamoDB(this, 'DynamoDBModuleStack', {
            DynamoDBIAMRole: iamModuleStack.DynamoDBRole,
        });
        utils.addTemplateUrl(
            dynamoDBModuleStack,
            Fn.join('', [
                'https://%%BUCKET_NAME%%-',
                Aws.REGION,
                '.s3.',
                Aws.URL_SUFFIX,
                '/%%SOLUTION_NAME%%/%%VERSION%%/msam-dynamodb-release.template',
            ]),
        );

        // Core Module Stack
        const coreModuleStack = new MsamCore(this, 'CoreModuleStack', {
            CoreIAMRole: iamModuleStack.CoreRole,
            AlarmsTableName: dynamoDBModuleStack.AlarmsTable,
            ChannelsTableName: dynamoDBModuleStack.ChannelsTable,
            ContentTableName: dynamoDBModuleStack.ContentTable,
            EventsTableName: dynamoDBModuleStack.EventsTable,
            LayoutTableName: dynamoDBModuleStack.LayoutTable,
            SettingsTableName: dynamoDBModuleStack.SettingsTable,
            CloudWatchEventsTableName: dynamoDBModuleStack.CloudWatchEventsTable,
            NotesTableName: dynamoDBModuleStack.NotesTable,
            CacheItemTTL: cacheItemTTL.valueAsString,
            RootStackName: Aws.STACK_NAME,
        });
        utils.addTemplateUrl(
            coreModuleStack,
            Fn.join('', [
                'https://%%BUCKET_NAME%%-',
                Aws.REGION,
                '.s3.',
                Aws.URL_SUFFIX,
                '/%%SOLUTION_NAME%%/%%VERSION%%/msam-core-release.template',
            ]),
        );

        // Events Module Stack
        const eventsModuleStack = new MsamEvents(this, 'EventsModuleStack', {
            AlarmsTableName: dynamoDBModuleStack.AlarmsTable,
            EventsTableName: dynamoDBModuleStack.EventsTable,
            ContentTableName: dynamoDBModuleStack.ContentTable,
            CloudWatchEventsTableName: dynamoDBModuleStack.CloudWatchEventsTable,
            ItemTTL: cacheItemTTL.valueAsString,
            EventsTableRegion: Aws.REGION,
            EventsIAMRole: iamModuleStack.EventsRole,
        });
        utils.addTemplateUrl(
            eventsModuleStack,
            Fn.join('', [
                'https://%%BUCKET_NAME%%-',
                Aws.REGION,
                '.s3.',
                Aws.URL_SUFFIX,
                '/%%SOLUTION_NAME%%/%%VERSION%%/msam-events-release.template',
            ])
        );

        // Browser App Module Stack
        const browserAppModuleStack = new MsamBrowserApp(this, 'BrowserAppModuleStack', {
            WebIAMRole: iamModuleStack.WebRole,
        });
        utils.addTemplateUrl(
            browserAppModuleStack,
            Fn.join('', [
                'https://%%BUCKET_NAME%%-',
                Aws.REGION,
                '.s3.',
                Aws.URL_SUFFIX,
                '/%%SOLUTION_NAME%%/%%VERSION%%/msam-browser-app-release.template',
            ]),
        );

        // Service Catalog - App Registry
        utils.applyAppRegistry(this, [
            {
                stackName: 'DynamoDB',
                stack: dynamoDBModuleStack,
            },
            {
                stackName: 'IAM',
                stack: iamModuleStack,
            },
            {
                stackName: 'Events',
                stack: eventsModuleStack,
            },
            {
                stackName: 'Browser',
                stack: browserAppModuleStack,
            },
            {
                stackName: 'Core',
                stack: coreModuleStack,
            },
        ]);

        /**
         * Outputs
         */
        utils.createCfnOutput(this, 'MSAMBrowserURL', {
            description: 'URL for the MSAM browser application',
            value: browserAppModuleStack.MSAMBrowserURL,
        });

        utils.createCfnOutput(this, 'EndpointURL', {
            description: 'The endpoint needed by the MSAM browser application',
            value: coreModuleStack.EndpointURL,
        });

        utils.createCfnOutput(this, 'APIKeyID', {
            description: 'Link for retrieving API key needed by the MSAM browser application',
            value: coreModuleStack.APIKeyID,
        });
    }
}
