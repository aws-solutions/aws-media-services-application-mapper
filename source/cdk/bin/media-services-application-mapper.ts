#!/usr/bin/env node
/*! Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
       SPDX-License-Identifier: Apache-2.0 */

import * as cdk from 'aws-cdk-lib';
import { MediaServicesApplicationMapper } from '../lib/media-services-application-mapper';
import { AwsSolutionsChecks } from 'cdk-nag';

const cdkApplication = new cdk.App();

(function createMediaServicesApplicationMapperStack(app) {
    return new MediaServicesApplicationMapper(app, 'MediaServicesApplicationMapper');
})(cdkApplication);

//cdk nag
cdk.Aspects.of(cdkApplication).add(new AwsSolutionsChecks({ verbose: true }));
