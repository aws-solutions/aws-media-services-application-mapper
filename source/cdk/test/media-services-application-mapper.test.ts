/*! Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
       SPDX-License-Identifier: Apache-2.0 */

import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { MediaServicesApplicationMapper } from '../lib/media-services-application-mapper';

test('media-services-application-mapper snapshot test', () => {
    const app = new cdk.App();
    const stack = new MediaServicesApplicationMapper(app, 'MediaServicesApplicationMapper');
    const template = Template.fromStack(stack);

    expect(template).toMatchSnapshot();
});
