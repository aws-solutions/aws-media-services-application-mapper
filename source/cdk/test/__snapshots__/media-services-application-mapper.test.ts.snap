// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`media-services-application-mapper snapshot test 1`] = `
{
  "Description": "(SO0048) Media Services Application Mapper Root Stack %%VERSION%%",
  "Mappings": {
    "AnonymizedData": {
      "SendAnonymizedData": {
        "Data": "Yes",
      },
    },
    "Solution": {
      "Data": {
        "AppRegistryApplicationName": "media-services-application-mapper",
        "ApplicationType": "AWS-Solutions",
        "ID": "SO0048",
        "SolutionName": "Media Services Application Mapper",
        "Version": "%%VERSION%%",
      },
    },
  },
  "Outputs": {
    "APIKeyID": {
      "Description": "Link for retrieving API key needed by the MSAM browser application",
      "Value": {
        "Fn::Join": [
          "",
          [
            "https://",
            {
              "Ref": "AWS::Region",
            },
            ".console.aws.amazon.com/apigateway/home?region=",
            {
              "Ref": "AWS::Region",
            },
            "#/api-keys/",
            {
              "Fn::GetAtt": [
                "CoreModuleStackNestedStackCoreModuleStackNestedStackResource2A4BEE47",
                "Outputs.MediaServicesApplicationMapperCoreModuleStackAPIKey9BCD1F3FRef",
              ],
            },
          ],
        ],
      },
    },
    "EndpointURL": {
      "Description": "The endpoint needed by the MSAM browser application",
      "Value": {
        "Fn::Join": [
          "",
          [
            "https://",
            {
              "Fn::GetAtt": [
                "CoreModuleStackNestedStackCoreModuleStackNestedStackResource2A4BEE47",
                "Outputs.MediaServicesApplicationMapperCoreModuleStackRestAPI418838E7Ref",
              ],
            },
            ".execute-api.",
            {
              "Ref": "AWS::Region",
            },
            ".",
            {
              "Ref": "AWS::URLSuffix",
            },
            "/msam/",
          ],
        ],
      },
    },
    "MSAMBrowserURL": {
      "Description": "URL for the MSAM browser application",
      "Value": {
        "Fn::Join": [
          "",
          [
            "https://",
            {
              "Fn::GetAtt": [
                "BrowserAppModuleStackNestedStackBrowserAppModuleStackNestedStackResource6F2C1DA7",
                "Outputs.MediaServicesApplicationMapperBrowserAppModuleStackMSAMAppBucketCloudFrontDistribution4829FDD0DomainName",
              ],
            },
          ],
        ],
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "CacheItemTTL": {
      "AllowedPattern": "\\S+",
      "ConstraintDescription": "Please enter a value for this field.",
      "Default": "86400",
      "Description": "This is the maximum time in seconds a cached item will remain if never updated (3600 = 1 hour, 86400 = 1 day, 604800 = 1 week).",
      "MinLength": 1,
      "Type": "String",
    },
  },
  "Resources": {
    "AppRegistryApplication": {
      "Properties": {
        "Description": "Service Catalog application to track and manage all your resources for the solution Media Services Application Mapper",
        "Name": {
          "Fn::Join": [
            "-",
            [
              {
                "Fn::FindInMap": [
                  "Solution",
                  "Data",
                  "AppRegistryApplicationName",
                ],
              },
              {
                "Ref": "AWS::Region",
              },
              {
                "Ref": "AWS::AccountId",
              },
              {
                "Ref": "AWS::StackName",
              },
            ],
          ],
        },
        "Tags": {
          "Solutions:ApplicationType": {
            "Fn::FindInMap": [
              "Solution",
              "Data",
              "ApplicationType",
            ],
          },
          "Solutions:SolutionID": {
            "Fn::FindInMap": [
              "Solution",
              "Data",
              "ID",
            ],
          },
          "Solutions:SolutionName": {
            "Fn::FindInMap": [
              "Solution",
              "Data",
              "SolutionName",
            ],
          },
          "Solutions:SolutionVersion": {
            "Fn::FindInMap": [
              "Solution",
              "Data",
              "Version",
            ],
          },
        },
      },
      "Type": "AWS::ServiceCatalogAppRegistry::Application",
    },
    "AppRegistryApplicationAttributeAssociation": {
      "Properties": {
        "Application": {
          "Fn::GetAtt": [
            "AppRegistryApplication",
            "Id",
          ],
        },
        "AttributeGroup": {
          "Fn::GetAtt": [
            "DefaultApplicationAttributes",
            "Id",
          ],
        },
      },
      "Type": "AWS::ServiceCatalogAppRegistry::AttributeGroupAssociation",
    },
    "AppRegistryApplicationNestedStackAssociationBrowser": {
      "Properties": {
        "Application": {
          "Fn::GetAtt": [
            "AppRegistryApplication",
            "Id",
          ],
        },
        "Resource": {
          "Ref": "BrowserAppModuleStackNestedStackBrowserAppModuleStackNestedStackResource6F2C1DA7",
        },
        "ResourceType": "CFN_STACK",
      },
      "Type": "AWS::ServiceCatalogAppRegistry::ResourceAssociation",
    },
    "AppRegistryApplicationNestedStackAssociationCore": {
      "Properties": {
        "Application": {
          "Fn::GetAtt": [
            "AppRegistryApplication",
            "Id",
          ],
        },
        "Resource": {
          "Ref": "CoreModuleStackNestedStackCoreModuleStackNestedStackResource2A4BEE47",
        },
        "ResourceType": "CFN_STACK",
      },
      "Type": "AWS::ServiceCatalogAppRegistry::ResourceAssociation",
    },
    "AppRegistryApplicationNestedStackAssociationDynamoDB": {
      "Properties": {
        "Application": {
          "Fn::GetAtt": [
            "AppRegistryApplication",
            "Id",
          ],
        },
        "Resource": {
          "Ref": "DynamoDBModuleStackNestedStackDynamoDBModuleStackNestedStackResourceB4888FC2",
        },
        "ResourceType": "CFN_STACK",
      },
      "Type": "AWS::ServiceCatalogAppRegistry::ResourceAssociation",
    },
    "AppRegistryApplicationNestedStackAssociationEvents": {
      "Properties": {
        "Application": {
          "Fn::GetAtt": [
            "AppRegistryApplication",
            "Id",
          ],
        },
        "Resource": {
          "Ref": "EventsModuleStackNestedStackEventsModuleStackNestedStackResourceB2BBE329",
        },
        "ResourceType": "CFN_STACK",
      },
      "Type": "AWS::ServiceCatalogAppRegistry::ResourceAssociation",
    },
    "AppRegistryApplicationNestedStackAssociationIAM": {
      "Properties": {
        "Application": {
          "Fn::GetAtt": [
            "AppRegistryApplication",
            "Id",
          ],
        },
        "Resource": {
          "Ref": "IAMModuleStackNestedStackIAMModuleStackNestedStackResource8BA6A78E",
        },
        "ResourceType": "CFN_STACK",
      },
      "Type": "AWS::ServiceCatalogAppRegistry::ResourceAssociation",
    },
    "AppRegistryApplicationStackAssociation": {
      "Properties": {
        "Application": {
          "Fn::GetAtt": [
            "AppRegistryApplication",
            "Id",
          ],
        },
        "Resource": {
          "Ref": "AWS::StackId",
        },
        "ResourceType": "CFN_STACK",
      },
      "Type": "AWS::ServiceCatalogAppRegistry::ResourceAssociation",
    },
    "BrowserAppModuleStackNestedStackBrowserAppModuleStackNestedStackResource6F2C1DA7": {
      "DeletionPolicy": "Delete",
      "DependsOn": [
        "IAMModuleStackNestedStackIAMModuleStackNestedStackResource8BA6A78E",
      ],
      "Properties": {
        "Parameters": {
          "referencetoMediaServicesApplicationMapperIAMModuleStackNestedStackIAMModuleStackNestedStackResourceF97053D6OutputsMediaServicesApplicationMapperIAMModuleStackWebRole1122C569Arn": {
            "Fn::GetAtt": [
              "IAMModuleStackNestedStackIAMModuleStackNestedStackResource8BA6A78E",
              "Outputs.MediaServicesApplicationMapperIAMModuleStackWebRole1122C569Arn",
            ],
          },
        },
        "TemplateURL": {
          "Fn::Join": [
            "",
            [
              "https://%%BUCKET_NAME%%-",
              {
                "Ref": "AWS::Region",
              },
              ".s3.",
              {
                "Ref": "AWS::URLSuffix",
              },
              "/%%SOLUTION_NAME%%/%%VERSION%%/msam-browser-app-release.template",
            ],
          ],
        },
      },
      "Type": "AWS::CloudFormation::Stack",
      "UpdateReplacePolicy": "Delete",
    },
    "CoreModuleStackNestedStackCoreModuleStackNestedStackResource2A4BEE47": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "Parameters": {
          "SendAnonymizedData": {
            "Fn::FindInMap": [
              "AnonymizedData",
              "SendAnonymizedData",
              "Data",
            ],
          },
          "referencetoMediaServicesApplicationMapperCacheItemTTL6DD8B4F7Ref": {
            "Ref": "CacheItemTTL",
          },
          "referencetoMediaServicesApplicationMapperDynamoDBModuleStackNestedStackDynamoDBModuleStackNestedStackResource588058D0OutputsMediaServicesApplicationMapperDynamoDBModuleStackAlarmsF6F9E932Ref": {
            "Fn::GetAtt": [
              "DynamoDBModuleStackNestedStackDynamoDBModuleStackNestedStackResourceB4888FC2",
              "Outputs.MediaServicesApplicationMapperDynamoDBModuleStackAlarmsF6F9E932Ref",
            ],
          },
          "referencetoMediaServicesApplicationMapperDynamoDBModuleStackNestedStackDynamoDBModuleStackNestedStackResource588058D0OutputsMediaServicesApplicationMapperDynamoDBModuleStackChannels0905569CRef": {
            "Fn::GetAtt": [
              "DynamoDBModuleStackNestedStackDynamoDBModuleStackNestedStackResourceB4888FC2",
              "Outputs.MediaServicesApplicationMapperDynamoDBModuleStackChannels0905569CRef",
            ],
          },
          "referencetoMediaServicesApplicationMapperDynamoDBModuleStackNestedStackDynamoDBModuleStackNestedStackResource588058D0OutputsMediaServicesApplicationMapperDynamoDBModuleStackCloudWatchEvents6ACDEB1ARef": {
            "Fn::GetAtt": [
              "DynamoDBModuleStackNestedStackDynamoDBModuleStackNestedStackResourceB4888FC2",
              "Outputs.MediaServicesApplicationMapperDynamoDBModuleStackCloudWatchEvents6ACDEB1ARef",
            ],
          },
          "referencetoMediaServicesApplicationMapperDynamoDBModuleStackNestedStackDynamoDBModuleStackNestedStackResource588058D0OutputsMediaServicesApplicationMapperDynamoDBModuleStackContent622D5AD1Ref": {
            "Fn::GetAtt": [
              "DynamoDBModuleStackNestedStackDynamoDBModuleStackNestedStackResourceB4888FC2",
              "Outputs.MediaServicesApplicationMapperDynamoDBModuleStackContent622D5AD1Ref",
            ],
          },
          "referencetoMediaServicesApplicationMapperDynamoDBModuleStackNestedStackDynamoDBModuleStackNestedStackResource588058D0OutputsMediaServicesApplicationMapperDynamoDBModuleStackEventsFE64F5A1Ref": {
            "Fn::GetAtt": [
              "DynamoDBModuleStackNestedStackDynamoDBModuleStackNestedStackResourceB4888FC2",
              "Outputs.MediaServicesApplicationMapperDynamoDBModuleStackEventsFE64F5A1Ref",
            ],
          },
          "referencetoMediaServicesApplicationMapperDynamoDBModuleStackNestedStackDynamoDBModuleStackNestedStackResource588058D0OutputsMediaServicesApplicationMapperDynamoDBModuleStackLayout9C4BCC32Ref": {
            "Fn::GetAtt": [
              "DynamoDBModuleStackNestedStackDynamoDBModuleStackNestedStackResourceB4888FC2",
              "Outputs.MediaServicesApplicationMapperDynamoDBModuleStackLayout9C4BCC32Ref",
            ],
          },
          "referencetoMediaServicesApplicationMapperDynamoDBModuleStackNestedStackDynamoDBModuleStackNestedStackResource588058D0OutputsMediaServicesApplicationMapperDynamoDBModuleStackResourceNotes46EADC5ERef": {
            "Fn::GetAtt": [
              "DynamoDBModuleStackNestedStackDynamoDBModuleStackNestedStackResourceB4888FC2",
              "Outputs.MediaServicesApplicationMapperDynamoDBModuleStackResourceNotes46EADC5ERef",
            ],
          },
          "referencetoMediaServicesApplicationMapperDynamoDBModuleStackNestedStackDynamoDBModuleStackNestedStackResource588058D0OutputsMediaServicesApplicationMapperDynamoDBModuleStackSettingsDBCDD26FRef": {
            "Fn::GetAtt": [
              "DynamoDBModuleStackNestedStackDynamoDBModuleStackNestedStackResourceB4888FC2",
              "Outputs.MediaServicesApplicationMapperDynamoDBModuleStackSettingsDBCDD26FRef",
            ],
          },
          "referencetoMediaServicesApplicationMapperIAMModuleStackNestedStackIAMModuleStackNestedStackResourceF97053D6OutputsMediaServicesApplicationMapperIAMModuleStackCoreRole5F5FCFE3Arn": {
            "Fn::GetAtt": [
              "IAMModuleStackNestedStackIAMModuleStackNestedStackResource8BA6A78E",
              "Outputs.MediaServicesApplicationMapperIAMModuleStackCoreRole5F5FCFE3Arn",
            ],
          },
        },
        "TemplateURL": {
          "Fn::Join": [
            "",
            [
              "https://%%BUCKET_NAME%%-",
              {
                "Ref": "AWS::Region",
              },
              ".s3.",
              {
                "Ref": "AWS::URLSuffix",
              },
              "/%%SOLUTION_NAME%%/%%VERSION%%/msam-core-release.template",
            ],
          ],
        },
      },
      "Type": "AWS::CloudFormation::Stack",
      "UpdateReplacePolicy": "Delete",
    },
    "DefaultApplicationAttributes": {
      "Properties": {
        "Attributes": {
          "applicationType": {
            "Fn::FindInMap": [
              "Solution",
              "Data",
              "ApplicationType",
            ],
          },
          "solutionID": {
            "Fn::FindInMap": [
              "Solution",
              "Data",
              "ID",
            ],
          },
          "solutionName": {
            "Fn::FindInMap": [
              "Solution",
              "Data",
              "SolutionName",
            ],
          },
          "version": {
            "Fn::FindInMap": [
              "Solution",
              "Data",
              "Version",
            ],
          },
        },
        "Description": "Attribute group for solution information",
        "Name": {
          "Fn::Join": [
            "-",
            [
              {
                "Ref": "AWS::Region",
              },
              {
                "Ref": "AWS::StackName",
              },
            ],
          ],
        },
      },
      "Type": "AWS::ServiceCatalogAppRegistry::AttributeGroup",
    },
    "DynamoDBModuleStackNestedStackDynamoDBModuleStackNestedStackResourceB4888FC2": {
      "DeletionPolicy": "Delete",
      "DependsOn": [
        "IAMModuleStackNestedStackIAMModuleStackNestedStackResource8BA6A78E",
      ],
      "Properties": {
        "Parameters": {
          "referencetoMediaServicesApplicationMapperIAMModuleStackNestedStackIAMModuleStackNestedStackResourceF97053D6OutputsMediaServicesApplicationMapperIAMModuleStackDynamoDBRole0562A431Arn": {
            "Fn::GetAtt": [
              "IAMModuleStackNestedStackIAMModuleStackNestedStackResource8BA6A78E",
              "Outputs.MediaServicesApplicationMapperIAMModuleStackDynamoDBRole0562A431Arn",
            ],
          },
        },
        "TemplateURL": {
          "Fn::Join": [
            "",
            [
              "https://%%BUCKET_NAME%%-",
              {
                "Ref": "AWS::Region",
              },
              ".s3.",
              {
                "Ref": "AWS::URLSuffix",
              },
              "/%%SOLUTION_NAME%%/%%VERSION%%/msam-dynamodb-release.template",
            ],
          ],
        },
      },
      "Type": "AWS::CloudFormation::Stack",
      "UpdateReplacePolicy": "Delete",
    },
    "EventsModuleStackNestedStackEventsModuleStackNestedStackResourceB2BBE329": {
      "DeletionPolicy": "Delete",
      "DependsOn": [
        "IAMModuleStackNestedStackIAMModuleStackNestedStackResource8BA6A78E",
      ],
      "Properties": {
        "Parameters": {
          "referencetoMediaServicesApplicationMapperCacheItemTTL6DD8B4F7Ref": {
            "Ref": "CacheItemTTL",
          },
          "referencetoMediaServicesApplicationMapperDynamoDBModuleStackNestedStackDynamoDBModuleStackNestedStackResource588058D0OutputsMediaServicesApplicationMapperDynamoDBModuleStackAlarmsF6F9E932Ref": {
            "Fn::GetAtt": [
              "DynamoDBModuleStackNestedStackDynamoDBModuleStackNestedStackResourceB4888FC2",
              "Outputs.MediaServicesApplicationMapperDynamoDBModuleStackAlarmsF6F9E932Ref",
            ],
          },
          "referencetoMediaServicesApplicationMapperDynamoDBModuleStackNestedStackDynamoDBModuleStackNestedStackResource588058D0OutputsMediaServicesApplicationMapperDynamoDBModuleStackCloudWatchEvents6ACDEB1ARef": {
            "Fn::GetAtt": [
              "DynamoDBModuleStackNestedStackDynamoDBModuleStackNestedStackResourceB4888FC2",
              "Outputs.MediaServicesApplicationMapperDynamoDBModuleStackCloudWatchEvents6ACDEB1ARef",
            ],
          },
          "referencetoMediaServicesApplicationMapperDynamoDBModuleStackNestedStackDynamoDBModuleStackNestedStackResource588058D0OutputsMediaServicesApplicationMapperDynamoDBModuleStackContent622D5AD1Ref": {
            "Fn::GetAtt": [
              "DynamoDBModuleStackNestedStackDynamoDBModuleStackNestedStackResourceB4888FC2",
              "Outputs.MediaServicesApplicationMapperDynamoDBModuleStackContent622D5AD1Ref",
            ],
          },
          "referencetoMediaServicesApplicationMapperDynamoDBModuleStackNestedStackDynamoDBModuleStackNestedStackResource588058D0OutputsMediaServicesApplicationMapperDynamoDBModuleStackEventsFE64F5A1Ref": {
            "Fn::GetAtt": [
              "DynamoDBModuleStackNestedStackDynamoDBModuleStackNestedStackResourceB4888FC2",
              "Outputs.MediaServicesApplicationMapperDynamoDBModuleStackEventsFE64F5A1Ref",
            ],
          },
          "referencetoMediaServicesApplicationMapperIAMModuleStackNestedStackIAMModuleStackNestedStackResourceF97053D6OutputsMediaServicesApplicationMapperIAMModuleStackEventsRole2AFF40D1Arn": {
            "Fn::GetAtt": [
              "IAMModuleStackNestedStackIAMModuleStackNestedStackResource8BA6A78E",
              "Outputs.MediaServicesApplicationMapperIAMModuleStackEventsRole2AFF40D1Arn",
            ],
          },
        },
        "TemplateURL": {
          "Fn::Join": [
            "",
            [
              "https://%%BUCKET_NAME%%-",
              {
                "Ref": "AWS::Region",
              },
              ".s3.",
              {
                "Ref": "AWS::URLSuffix",
              },
              "/%%SOLUTION_NAME%%/%%VERSION%%/msam-events-release.template",
            ],
          ],
        },
      },
      "Type": "AWS::CloudFormation::Stack",
      "UpdateReplacePolicy": "Delete",
    },
    "IAMModuleStackNestedStackIAMModuleStackNestedStackResource8BA6A78E": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "TemplateURL": {
          "Fn::Join": [
            "",
            [
              "https://%%BUCKET_NAME%%-",
              {
                "Ref": "AWS::Region",
              },
              ".s3.",
              {
                "Ref": "AWS::URLSuffix",
              },
              "/%%SOLUTION_NAME%%/%%VERSION%%/msam-iam-roles-release.template",
            ],
          ],
        },
      },
      "Type": "AWS::CloudFormation::Stack",
      "UpdateReplacePolicy": "Delete",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
